__tests__/integration/transaction.test.js

```diff
@@ -50,4 +50,51 @@ describe('transaction routes', () => {
     expect(response.body).toHaveProperty('version', tx.version);
     done();
   });
+
+  it('test confirmation blocks', async done => {
+    // Generates a transaction
+    const tx = await wallet1.injectFunds(1);
+    const txId = tx.hash;
+
+    // Queries for this transaction
+    const response = await TestUtils.request
+      .get('/wallet/tx-confirmation-blocks')
+      .query({
+        id: txId
+      })
+      .set({ 'x-wallet-id': wallet1.walletId });
+
+    expect(response.status).toBe(200);
+
+    // We might have some blocks while we are getting these requests
+    // so it's not easy to do an assert here
+    const height = await TestUtils.getFullNodeNetworkHeight();
+    const txData1 = await TestUtils.getFullNodeTransactionData(txId);
+
+    let firstBlockHeight = txData1.meta.first_block_height;
+    if (firstBlockHeight === null) {
+      // With this await here we know that we have at least one block confirming it
+      firstBlockHeight = Math.min(await TestUtils.waitNewBlock(height), height + 1);
+    }
+
+    const txData2 = await TestUtils.getFullNodeTransactionData(txId);
+    expect(txData2.meta.first_block_height).toBe(firstBlockHeight);
+
+    await TestUtils.waitNewBlock(firstBlockHeight);
+
+    const response2 = await TestUtils.request
+      .get('/wallet/tx-confirmation-blocks')
+      .query({
+        id: txId
+      })
+      .set({ 'x-wallet-id': wallet1.walletId });
+
+    const newHeight = await TestUtils.getFullNodeNetworkHeight();
+    // With the async requests, new blocks may be mined, so we can't assert the equal
+    // We will have at least the first block confirming plus one (with the await above)
+    expect(response2.body.confirmationNumber).toBeGreaterThan(0);
+    expect(response2.body.confirmationNumber).toBeLessThanOrEqual(newHeight - firstBlockHeight);
```

Comment:
The `less than` case would be if a new block is mined during the call to `getFullNodeNetworkHeight`?

Reply:
Yeah, exactly